File: zuzubibi/2월/2월 6일/폰켓몬.py
def solution(nums):
    l = len(nums)//2
    re = len(set(nums))
    # print(set(nums), l)
    return l if re > l else re

def anotherSolution(nums):
    return min(len(nums)/2, len(set(nums)))

print(solution([3,1,2,3])) # 2
print(solution([3,3,3,2,2,4])) # 3
print(solution([3,3,3,2,2,2])) # 2

File: wooseongshin/2월/6일/폰켓몬.java
import java.util.*;
class Solution {
    public int solution(int[] nums) {
        HashSet<Integer> set = new HashSet<>();
        for(int i = 0; i < nums.length; i++) {
            set.add(nums[i]);
        }

        int answer = set.size();
        if(answer > nums.length / 2) {
            answer = nums.length / 2;
        }

        return answer;
    }
}
